microsoft/testsuites/network/netinterface.py

```diff
@@ -0,0 +1,119 @@
+            3. Change nic state to up and down multiple times.
+                Each time after "up" state, verify ip address is assigned to nic
+                and internet is accessible via nic.
+
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            min_nic_count=1,
+            network_interface=schema.NetworkInterfaceOptionSettings(
+                data_path=schema.NetworkDataPath.Synthetic,
+            ),
+        ),
+    )
+    def validate_network_interface_ifup_ifdown(self, node: Node, log: Logger) -> None:
+        self._validate_netvsc_built_in(node)
+        network_interface_feature = node.features[NetworkInterface]
+
+        # Test loading and unloading netvsc driver
+        test_count = 0
+        while test_count < self.RELOAD_NETVSC_TEST_COUNT:
+            test_count += 1
+            # Unload hv_netvsc
+            network_interface_feature.reload_netvsc()
+
+        # Clean all dhclient processes, get IP & try ping
+        node_nic_info = Nics(node)
+        node_nic_info.initialize()
+        default_nic = node_nic_info.default_nic
+        default_route = node_nic_info.default_nic_route
+
+        node_nic_info.change_nic_state(default_nic, "up")
+        if not node_nic_info.nics[default_nic].ip_addr:
+            node.execute("kill $(pidof dhclient)", shell=True, sudo=True)
+            dhclient = node.tools[Dhclient]
+            dhclient.release()
+            dhclient.renew()
+            sleep(15)
+
+        self._ping_check
+
+        test_count = 0
+        while test_count < self.NET_IFUP_IFDOWN_TEST_COUNT:
+            self.__log.info(f"NET IFUP IFDOWN Test Iteration : {test_count}")
+            node_nic_info.change_nic_state(default_nic, "down")
+            node_nic_info.change_nic_state(default_nic, "up")
+            if not node_nic_info.default_nic:
+                # Add default route if missing after running ip link down/up
+                node.execute(f"ip route add {default_route}", shell=True, sudo=True)
+
+            if not node_nic_info.nics[default_nic].ip_addr:
+                node.execute("kill $(pidof dhclient)", shell=True, sudo=True)
+                dhclient = node.tools[Dhclient]
+                dhclient.release()
+                dhclient.renew()
+                sleep(15)
+
+            self._ping_check
+
+    def _validate_netvsc_built_in(self, node: Node) -> None:
+        uname_tool = node.tools[Uname]
+        kernel_ver = uname_tool.get_linux_information().kernel_version
+        config_path = f"/boot/config-{kernel_ver}"
+        if isinstance(node.os, CoreOs):
+            config_path = f"/usr/boot/config-{kernel_ver}"
+
+        netvsc_builtin_result = node.execute(
+            f"grep CONFIG_HYPERV_NET=y {config_path}", shell=True
+        )
+        if netvsc_builtin_result.exit_code == 0:
+            SkippedException("Skipping test since hv_netvsc module is built-in")
+
+    def _ping_check(self, node: Node) -> None:
```

Comment:
This step looks not necessary. If the VM cannot reach the internet, we are not able to connect from SSH.

Reply:
I think it is possible sometimes to ssh to VM even when connection to internet from VM doesn't work.

Reply:
It's possible, if the VM is tested in an internal network. But it doesn't related to the test cases. The test cases don't require internet access.